components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/dao/impl/AuthenticatorManagementDAOImpl.java

```diff
@@ -0,0 +1,266 @@
+import org.wso2.carbon.identity.application.common.constant.AuthenticatorMgtSQLConstants.Query;
+import org.wso2.carbon.identity.application.common.dao.AuthenticatorManagementDAO;
+import org.wso2.carbon.identity.application.common.exception.AuthenticatorMgtException;
+import org.wso2.carbon.identity.application.common.exception.AuthenticatorMgtServerException;
+import org.wso2.carbon.identity.application.common.model.Property;
+import org.wso2.carbon.identity.application.common.model.UserDefinedLocalAuthenticatorConfig;
+import org.wso2.carbon.identity.application.common.util.AuthenticatorMgtExceptionBuilder.AuthenticatorMgtError;
+import org.wso2.carbon.identity.base.AuthenticatorPropertyConstants.AuthenticationType;
+import org.wso2.carbon.identity.base.AuthenticatorPropertyConstants.DefinedByType;
+import org.wso2.carbon.identity.core.util.IdentityDatabaseUtil;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.wso2.carbon.identity.application.common.util.AuthenticatorMgtExceptionBuilder.buildServerException;
+
+/**
+ * This class implements the AuthenticatorManagementDAO interface which perform CRUD operation on database.
+ */
+public class AuthenticatorManagementDAOImpl implements AuthenticatorManagementDAO {
+
+    public static final String IS_TRUE_VALUE = "1";
+    public static final String IS_FALSE_VALUE = "0";
+    public static final String LOCAL_IDP_NAME = "LOCAL";
+
+    @Override
+    public UserDefinedLocalAuthenticatorConfig addUserDefinedLocalAuthenticator(
+            UserDefinedLocalAuthenticatorConfig authenticatorConfig, int tenantId) throws AuthenticatorMgtException {
+
+        NamedJdbcTemplate jdbcTemplate = new NamedJdbcTemplate(IdentityDatabaseUtil.getDataSource());
+        try {
+            int authenticatorConfigID = jdbcTemplate.withTransaction(template ->
+                template.executeInsert(Query.ADD_AUTHENTICATOR_SQL,
+                    statement -> {
+                        statement.setString(Column.NAME, authenticatorConfig.getName());
+                        statement.setString(Column.DISPLAY_NAME, authenticatorConfig.getDisplayName());
+                        statement.setString(Column.DEFINED_BY, authenticatorConfig.getDefinedByType().toString());
+                        statement.setString(Column.AUTHENTICATION_TYPE, authenticatorConfig.getAuthenticationType()
+                                .toString());
+                        statement.setString(Column.IS_ENABLED, String.valueOf(authenticatorConfig.isEnabled() ? 1 : 0));
+                        statement.setString(Column.IDP_NAME, LOCAL_IDP_NAME);
+                        statement.setInt(Column.TENANT_ID, tenantId);
+                    }, null, true));
+
+            if (authenticatorConfigID == 0) {
+                getAuthenticatorEntryId(authenticatorConfig.getName(), tenantId);
```

Comment:
```suggestion
               authenticatorConfigID = getAuthenticatorEntryId(authenticatorConfig.getName(), tenantId);
```
shouldn't we assign the value to the variable ?